packages/bigquery/src/dataset.js

```diff
@@ -70,6 +78,11 @@ function Dataset(bigQuery, id) {
      *
      * @example
      * dataset.exists(function(err, exists) {});
+     *
+     * //-
+     * // If the callback is omitted, we'll return a Promise.
+     * //-
+     * dataset.exists().then(function(exists) {});
```

Comment:
I'm unsure about the inconsistency in sometimes returning an array vs. the unboxed response. I don't know if we should needlessly arrayify single-arg responses, but then again, maybe that's expected / better for consistency? Curious on your thoughts!

// @jmdobry